src/MetadataUtility/Metadata/FrontierLabs/LogFileExtractor.cs

```diff
@@ -0,0 +1,64 @@
+// <copyright file="LogFileExtractor.cs" company="QutEcoacoustics">
+// All code in this file and all associated files are the copyright and property of the QUT Ecoacoustics Research Group.
+// </copyright>
+
+namespace MetadataUtility.Metadata.FrontierLabs
+{
+    using System.Threading.Tasks;
+    using MetadataUtility.Models;
+    using Microsoft.Extensions.Logging;
+
+    public class LogFileExtractor : IMetadataOperation
+    {
+        public const int MetadataOffset = 39;
+        private readonly ILogger<LogFileExtractor> logger;
+
+        public LogFileExtractor(ILogger<LogFileExtractor> logger)
+        {
+            this.logger = logger;
+        }
+
+        public ValueTask<bool> CanProcessAsync(TargetInformation information)
+        {
+            var result = information.HasBarltSupportFile();
+
+            return ValueTask.FromResult(result);
+        }
+
+        public ValueTask<Recording> ProcessFileAsync(TargetInformation information, Recording recording)
+        {
+            string[] lines = information.FileSystem.File.ReadAllLines(information.KnownSupportFiles["Log file"]);
+
+            int i = 0;
+
+            while (i < lines.Length)
+            {
+                if (lines[i].Contains("SD Card :"))
+                {
+                    break;
+                }
```

Comment:
BAR LTs support up to 4 SD cards

Reply:
This also means you won't know which card belongs to which recording unless you correlate them some how?

Reply:
It seems like new log files are created for each SD card, but couldn't find anything in the Frontier Labs documentation to confirm this. Instead I collect every instance of an SD card in a log file. If they aren't all representing the same SD Card, then I correlate them if possible (only works with newer firmware, where the filename is printed each time a new recording is started).